### ISSUE TITLE
Allow for testing of parallel plugin/sidecar update PRs

### ISSUE THREAD

**Issue by tsmaeder** [eclipse-che/che#18806]
### Is your enhancement related to a problem? Please describe.
When the update of a vscode extensions in a plugin requires an updated sidecar (https://github.com/eclipse/che-plugin-registry/pull/794, for example), there is no easy way to test that combination: the plugin registry built as part of the PR check references an image that does not exist, in this case. 
The current solution is to update the sidecar first and then update the plugin in a separate PR. However, in the case of the mentioned PR, the updated sidecar does not work with the old extension, thus, we would have a time window where workspaces using the updated sidecar would fail.

### Describe the solution you'd like
I would updated like to be able to test the PR with changes to both the sidecar and plugin. Merging should promote both changes simultaneously to "production".

**Comment by ericwill** [comment]
This actually used to work, before we started checking sidecar validity in the plugin registry build process. Since we turned that on, updates to both sidecars and plugins in the same PR will fail, because the tag is based on the SHA1 of the change: https://github.com/eclipse/che-plugin-registry/pull/848#issuecomment-783670984

A few options:
* we could build and push the sidecar tag during the sidecar PR build, but this would require the plugin build depending on the sidecar build/push to finish.
* we could skip checking of images if the SHA1 ID in the PR matches the tag being checked

cc @benoitf

**Comment by benoitf** [comment]
There are pros and cons on both sides. If we skip it means published version on surge is also unsuitable.
And we can't publish from forked repositories due to token issues so difficult to say.
Should we just abort and say side-cars need to be updated first ( nicer error message than today)
